Migrate to r

PavelBal opened the pull request:
Fixes #8

- Migrate to R
- Snapshot updated with version 12

Report for review: https://github.com/PavelBal/Amikacin-Model/blob/review/evaluation/report/report.md

PavelBal committed:
- Adjust evaluation plan to new framework
- Add Github workflows
- Remove Matlab script and add R Script

PavelBal committed:
- Re-create model snapshot with PK-Sim v12

PavelBal committed:
Rename snapshot file to stay consistent with the "xx-Model.json" naming

PavelBal commented:
@Yuri05 Why are no actions triggered in this PR? In others (e.g. https://github.com/Open-Systems-Pharmacology/Ethinylestradiol-Model/pull/12) they are.

PavelBal commented:
@Yuri05 what is the best (== easiest) way to create the report and compare it with the old one?

Yuri05 commented:
> @Yuri05 what is the best (== easiest) way to create the report and compare it with the old one?

I usually create a fork of the model repo, clone it locally and create a **review** branch, create the report locally and push it into the forked repo (e.g. https://github.com/Yuri05/Erythromycin-Model/tree/review/Evaluation/report)
Also I have a fork of the PBPK-Model-Library repo, where I create a branch per model report; copy the newly created report here and create a draft PR into PBPK-Model-Library repo (e.g. https://github.com/Open-Systems-Pharmacology/OSP-PBPK-Model-Library/pull/128).

For the local installation, if you already have ospsuite-R and ospsuite 12 installed, you need 
* the RE package: https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/releases/latest
* the QualificationRunner: https://github.com/Open-Systems-Pharmacology/QualificationRunner/releases/latest

PavelBal commented:
That's not easy :D I was hoping to somehow use the new workflow that you presented last time=

PavelBal commented:
@Yuri05 trying to run the workflow for this evaluation report I get the following error regarding populations:

<html>
<body>
<!--StartFragment--><pre tabindex="0" aria-label="Console Output" role="document" class="GNTYACMBB4B" id="rstudio_console_output" style="font-family: &quot;Lucida Console&quot;, monospace; font-size: 13.3333px; outline: none; border: none; word-break: break-all; margin: 0px; user-select: text; white-space: pre-wrap !important; line-height: 16px; color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: -webkit-left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span tabindex="-1" class="GNTYACMBC4B" role="document" style="outline: none;"><span class="GNTYACMBI4B ace_keyword" style="color: blue; white-space: pre; user-select: text;">&gt; </span><span class="GNTYACMBH3B ace_keyword" style="color: blue;">createQualificationReport(qualificationRunnerFolder = "C:/Program Files/Open Systems Pharmacology/QualificationRunner 12.0.1",
</span><span class="GNTYACMBI4B ace_keyword" style="color: blue; white-space: pre; user-select: text;">+ </span><span class="GNTYACMBH3B ace_keyword" style="color: blue;">                          createWordReport = FALSE)
</span><span class="GNTYACMBB4B" style="outline: none; border: none; word-break: break-all; margin: 0px; user-select: text; white-space: pre-wrap !important;">info: QualificationRunner[0]
      Starting qualification workflow
info: QualificationRunner[0]
      Starting validation runs...
info: QualificationRunner[0]
      Starting qualification runs...
info: QualificationRunner[0]
      Qualification scenario finished in 00m:41s:812ms
info: QualificationRunner[0]
      Qualification Workflow finished
[1] "Qualification Runner Duration: 0.7 minutes"
16/04/2025 - 16:17:44
i Info	Loading Simulations onto Qualification Workflow
  |                                                                                                                                                                                                         |   0%
Loading project 'Amikacin', and simulation 'Walker 1979 Volunteer 6.'
  |===================================================================                                                                                                                                      |  33%
Loading project 'Amikacin', and simulation 'Walker 1979 Volunteer 9.'
  |======================================================================================================================================                                                                   |  67%
Loading project 'Amikacin', and simulation 'Walker 1979 POP.'
16/04/2025 - 16:17:46
! Warning	NA in coercion to boolean
	
&gt; Error Trace
	 ↪ createQualificationReport(qualificationRunnerFolder = "C:/Program Files/Open Systems Pharmacology/QualificationRunner 12.0.1", 
	  ↪ loadQualificationWorkflow(workflowFolder = reOutputFolder, configurationPlanFile = configurationPlanFile)
	   ↪ PopulationSimulationSet$new(simulationSetName = simulationSetName, 
	    ↪ initialize(...)
	     ↪ populationName %||% trimFileName(populationFile, extension = "csv")
	      ↪ trimFileName(populationFile, extension = "csv")
	       ↪ sub(pattern = paste0("[.].*", extension), "[.].*$", replacement = "", 
</span><div class="GNTYACMBNID ace_console_error" style="position: relative; overflow: hidden; margin: 5px 5px 5px 0px; padding: 5px 0px 5px 10px; white-space: pre-line; background-color: rgb(230, 230, 230);"><div class="GNTYACMBOID" style="top: 5px; float: right; margin-right: 10px; margin-left: 5px; margin-bottom: 15px; line-height: 10.6667px; position: relative; user-select: none;"><div><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABIAAAASCAYAAABWzo5XAAAAfElEQVR4XmNgwAXWv8sEY4rA2vd+DOve/wFjEJsssOGjOdCA/ygYJEYS2PhRBcMQGAbJEQVWfRbF0IyOQWrwgs3PuDA04cIgtVjBqv/MGIoJYZAeDICuiFhMEKBrIFojOkA3YAQYhAvjBOgKCWGcAF0hIYwToCskhActAAATrGcoccG/iQAAAABJRU5ErkJggg==" width="18" height="18" class="GNTYACMBPID show_traceback_image" alt="" aria-hidden="true" style="cursor: pointer; width: 9px !important; height: 9px !important;"> <a class="GNTYACMBAJD show_traceback_text" style="font-family: &quot;Segoe UI&quot;, sans-serif; font-size: 10.6667px; opacity: 0.5; cursor: pointer;">Show Traceback</a></div><div><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABIAAAASCAYAAABWzo5XAAABZ0lEQVR4Xp2Uv0rEQBDGVxtrNZuzECvRRgsVBCtLbXwL8QnsfQ4fQBThsomFoCBYKIh/Wguvze7eRRQUq+O8OJOQzd3cJKIffEUy8/uYJbMRokJTzXjWV+ZQKqulMl3wtRfYTdpXq2ll1wF8B6fEPS/Uu7Sf1eSZngMgYUIKd/0g3qCc8Jp6VRyk4+5ZGcXA1A8uANgsA15eytDcNML2soz0CgOxxuM3gvYSspghRJqOwRR7Mj/OPQWq7Cl9hwyygycSftRpQOGDAjVO5Gky4wKcIBWKPQao8usQzzT8yy4QF5AWf/H3wDyl5s9bE1DsMwBr2Po3muEE16FFgRo/UT5T/tX+FPSCTJlQ7lEHxj1igArbE2SyPYIM/GoX4Fvc7Gwypa9GoRE/47pktwFYiZsNiWtDdy2IF6DwxcCF+35ottyJ8K5hBicZmO2KMAix+7S/Vl6kF2VojwH+lPnGP8K/aIf2FfoBOjBEnzMy4qIAAAAASUVORK5CYII=" width="18" height="18" class="GNTYACMBPID rerun_image" alt="" style="cursor: pointer; width: 9px !important; height: 9px !important;"> <a class="GNTYACMBAJD rerun_text" style="font-family: &quot;Segoe UI&quot;, sans-serif; font-size: 10.6667px; opacity: 0.5; cursor: pointer;">Rerun with Debug</a></div></div><div class="GNTYACMBBJD GNTYACMBL3B  ace_constant" style="white-space: pre; display: inline; overflow-wrap: break-word; word-break: break-word; color: rgb(197, 6, 11);"><span>Error: NA in coercion to boolean</span></div><div class="GNTYACMBCJD stack_trace" aria-hidden="true" style="margin-top: 0.5em; margin-left: 0.3em; display: block;">
8. | stop(errorCondition$message, call. = FALSE)
-- | --


</div></div><span></span></pre><br class="Apple-interchange-newline"><!--EndFragment-->
</body>
</html>

Can you try to run it?

Yuri05 commented:
@PavelBal Worked for me. Which package versions are you using? I am using RE 2.2.369 (latest released)
```
	R version: R version 4.2.2 (2022-10-31 ucrt)
	OSP Suite Package version: 12.1.0
	OSP Reporting Engine version: 2.2.369
	tlf version: 1.6.174
```

PavelBal commented:
This happens here:

https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/blob/344b0876623f2dd43295f6fd3ae60d7cbe8997e7/R/utils.R#L78

If I run this code:

```
extension <- "csv"
fileName <- "Walker 1979 POP-Population.csv"
sub(
      pattern = paste0("[.].*", extension),
      "[.].*$",
      replacement = "",
      x = fileName
    )
```

I get the error:

```
Error in sub(as.character(pattern), as.character(replacement), x, ignore.case,  : 
  NA in coercion to boolean
```

I have R 4.5.0. Can you run this specific code?

PavelBal commented:
It looks like 4.5.0 changed the way unnamed arguments are used. In the code above, I think there is a bug because this 

https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/blob/344b0876623f2dd43295f6fd3ae60d7cbe8997e7/R/utils.R#L87

should not be there at all. In R 4.4 this was ignored (??), but in 4.5.0  the first unnamed argument is assigned to the first unused argument of the function, or so... So ` "[.].*$"` is assigned to the argument `ignore.case`, which is expected to be a boolean.

PavelBal commented:
@Yuri05 ready for review

Still don't know why the actions are not triggered.

PavelBal committed:
Update workflows

Commit:
Fixed typo in evaluation_plan.json
Mapped observed data in Amikacin-Model.json
Removed obsolete GH Action XRefCheck.yml

Commit:
Renamed folder "evaluation" => "Evaluation"
Adjusted wordlist.txt

Commit:
Fix ValidateEvaluationPlan.yml

Yuri05 commented:
Will approve once https://github.com/PavelBal/Amikacin-Model/pull/1 was merged.
Report comparison: https://github.com/Open-Systems-Pharmacology/OSP-PBPK-Model-Library/pull/136

PavelBal committed:
Merge pull request #1 from Yuri05/migrate-to-r

Fixes
